.github/workflows/playwright.yml

```diff
@@ -143,62 +148,18 @@ jobs:
     - uses: actions/upload-artifact@v4
       if: steps.git-check.outputs.modified == 'true'
       with:
-        name: playwright-report-ubuntu-${{ github.sha }}
+        name: playwright-report-ubuntu-${{ matrix.shardIndex }}-${{ github.sha }}
         path: playwright-report/
         retention-days: 30
     # if have previous run results, use them
```

Comment:
Yeah yeah I'll explain what it does and request a change.

So what this does is it tries to download the report from the last run for this exact commit i.e. if someone has re-run this job it will get the artifact.

And I'm thinking we probably want to re-introduce the logic we had before we did our "roll our own retries"

i.e. what got remove in this PR https://github.com/KittyCAD/modeling-app/pull/3002/files

<img width="692" alt="image" src="https://github.com/user-attachments/assets/32777733-3ed7-45d8-b6c8-a64451f30306">

That is it check if there is a `.last-run.json` and if it exists than run with `--last-failed` if not run normally, this make just make retries run much quicker

I do like what you've got with the [single retry](https://github.com/KittyCAD/modeling-app/pull/3199/files#r1701725715), I think that's good to have as well.

Does that all make sense?

Reply:
no probs, I will bring it back